api/v1/controllers/EventController.js

```diff
@@ -0,0 +1,154 @@
+                        next();
+                        return null;
+                });
+}
+
+function getLocations(req, res, next) {
+        return services.EventService.getLocations()
+                .then(function (result) {
+                        res.body = result.length == 0 ? [] : result.toJSON();
+                        next();
+                        return null;
+                });
+}
+
+function addLocationToEvents(req, res, next) {
+        return services.EventService.addLocationToEvents(req.body.eventIds,
+                                                         req.body.locationId)
+                .then(function (event) {
+                        next();
+                        return null;
+                })
+                .catch(function (err) {
+                        next(err);
+                        return null;
+                });
+}
+
+function addEventToLocations(req, res, next) {
+        return services.EventService.addEventToLocations(req.body.eventId,
+                                                         req.body.locationIds)
+                .then(function (result) {
+                        return services.EventService.findEventById(req.body.eventId);
+                })
+                .then(function (event) {
+                        res.body = event.toJSON();
+                        next();
+                        return null;
+                })
+                .catch(function (err) {
+                        next(err);
+                        return null;
+                });
+}
+
+function deleteReference(req, res, next) {
+        if (!req.body.eventIds && !req.body.locationIds)
+        {
+                var message = "Either eventIds or locationIds must be non-null";
+                var source = "eventIds and locationIds";
+                next(new errors.InvalidParameterError(message, source));
+                return null;
+        }
+        return services.EventService.deleteReference(req.body.eventIds, req.body.locationIds)
+                .then(function (result) {
+                        next();
+                        return null;
+                });
+}
+
+function deleteEvent(req, res, next) {
+        return services.EventService.deleteEvent(req.body.eventId)
+                .then(function (result) {
+                        next();
+                        return null;
+                });
+}
+
+router.use(bodyParser.json());
+
+router.get('/', getEvents);
+router.get('/locations', getLocations);
+router.post('/create', middleware.auth,
+            middleware.permission(utils.roles.ORGANIZERS), // Change to SUPERUSER?
+            middleware.request(requests.EventCreationRequest),
+            createEvent);
+router.post('/delete', middleware.auth,
+            middleware.permission(utils.roles.ORGANIZERS),
+            middleware.request(requests.DeleteEventRequest),
+            deleteEvent);
+router.post('/deleteReference', middleware.auth,
```

Comment:
Additionally, I'm going to really question when we'd ever use this and if it should be in here. This seems like a dangerous endpoint to have, but also not necessary.